src/workers/background-proc/core.ts

```diff
@@ -124,6 +131,68 @@ export class BackgroundCore {
     await queue.onIdle()
   }
 
+  async getVideoSize(url: string): Promise<VideoSize> {
+    const text = await Axios.get(url);
+    const textData = text.data;
+    console.log(textData);
+    let regex = /RESOLUTION=(.+),/g;
+    let found = textData.match(regex);
+    if (found === null || found.length === 0) {
+      regex = /RESOLUTION=(.+)\n/g;
+      found = textData.match(regex);
+    }
+    const size = found[0].replace('RESOLUTION=', '').replace(',', '').replace('\n', '');
+    console.log(`Size ${size}`);
+    const sizeComps = size.split(`x`);
+    return {
+      width: parseInt(sizeComps[0]),
+      height: parseInt(sizeComps[1]),
+    }
+  }
+
+  async updateHeightWeight() {
+    const PQueue = (await import('p-queue')).default
+    const queue = new PQueue({ concurrency: 50 })
+
+    // Step 1. Find all posts having 3speak as app name and height-width is null (not updated)
+    const items = await this.posts.find({
+      parent_author: "",
+      "json_metadata.app": {
+        '$regex': '3speak'
+      }, 
+      height: { $eq: null },
+      width: { $eq: null },
+    }, {
+      limit: 2000
+    })
+
+    // Step 2. Find Play URL for each item, and get video size - height, width & update it.
+    for await (let itm of items) {
+      queue.add(async () => {
+        // Step 2.1. get possible play url
+        let play_url = (itm.json_metadata?.video?.info?.sourceMap || []).find(
+          (e) => e.type === 'video',
+        )?.url
+        play_url = play_url
+          ? play_url
+          : `https://threespeakvideo.b-cdn.net/${itm.permlink}/default.m3u8`
+        if (play_url.startsWith('ipfs://')) {
+          play_url = `https://ipfs-3speak.b-cdn.net/ipfs/${play_url.replace("ipfs://","")}`;
+        }
+        // Step 2.2 get video size.
+        const size = await this.getVideoSize(play_url);
+
+        // Step 2.3 Update in the db
+        await this.posts.findOneAndUpdate(
+          { _id: itm._id }, 
+          { $set: { 'height': size.height, 'width': size.width } }
```

Comment:
This might go better under a nested field rather than being at top level. 
`app_metadata.spkvideo` would be a good place to put it